### ISSUE TITLE
Should we specify what signature/ZKP schemes a verifier supports?

### ISSUE THREAD

**Issue by csuwildcat** [decentralized-identity/proof-presentation#9]
It feels like leaving the verifier to wade through a schema definition or 'scheme-sniff' returned data for the crypto schemes the response items use would be messy, misfire-prone process. As such, should we consider adding a field to the Proof Presentation where the verifier can state what schemes they support verification of?

**Comment by David-Chadwick** [comment]
This is a good point. Currently I am assuming that JWT, JSON-LD proofs and ZKP will be discreet islands that dont interwork, and that even within one of these islands there may not be complete interworking due to the numerous options that are available

**Comment by OR13** [comment]
@tplooker @csuwildcat any update on this? how do we move this forward?

**Comment by tplooker** [comment]
I agree, we in general need the ability between all actors in a verifiable credential ecosystem for them to communicate to one and other, what cryptography they support (e.g what algorithms I support signing and verifying), otherwise we will create headaches around achieving crypto-graphic agility in future.

I think a place for a verifier to express what cryptographic algorithms they support in verifying assertions is wise. But the same applies for an issuer and holder being able to advertise what algorithms they support.

**Comment by tplooker** [comment]
This expands beyond just ZKP schemes too, there are different algorithms for even more traditional digital signatures.

**Comment by OR13** [comment]
I'm having trouble seeing the discussion here as much more than "we wish things were clearer" :) can we provide an example of an improvement?

**Comment by csuwildcat** [comment]
Will tackle in future release of spec.
